Updates changelog to include missed releases

TheLonelyGhost opened the pull request:
Looks like we haven't been too diligent on updating the changelog for each release. Now all version changes (except one flub with `2.6.14` -> `2.6.1` -> `2.7.0`) are [tagged in git](https://github.com/TranscribersOfReddit/TranscribersOfReddit/tags).

This PR is a followup by attempting to describe the changes between `v2.7.1` -> `v3.0.0` and between `v3.0.0` -> `v3.0.1`

Also prunes stale `console_scripts` in `setup.py`.

NGTmeaty commented:
@TheLonelyGhost could we also update the readme in this pull?

TheLonelyGhost commented:
NO! ^_^

... sure, gimme a sec

TheLonelyGhost commented:
@NGTmeaty Is that what you wanted? What else do you see in the README that needs to be updated?

itsthejoker commented:
I think he was referring more specifically to the information about the other bots which is no longer valid.

TheLonelyGhost commented:
You know, I forgot we followed through with splitting up the bots into separate repositories. I removed the entries in `setup.py` so it's less confusing and updated associated docs (changelog included).

Is that what we're talking about with updating the readme?

NGTmeaty commented:
looks good :+1:
